### PR TITLE
Disable old deprecated formula

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Generated using:
```bash
#!/usr/bin/env bash

if ! formulae=$(curl --fail -s https://formulae.brew.sh/api/formula.json | jq --compact-output ".[] | select(.deprecated and .deprecation_date != null) 
| {date: @text \"\(.deprecation_date)T00:00:00Z\" | fromdateiso8601, name: .name}"); then
  echo "Fail!"
  exit 1
fi

date=$(date +"%Y-%m-%d")
year_ago=$(date -v '-1y' +"%s")
core_repo=$(brew --repo homebrew/core)
for formula in $formulae; do
  name=$(jq -r '.name' <<< "$formula")
  if [ "$year_ago" -lt $(jq -r '.date' <<< "$formula") ]; then
    printf "deprecation in %s is too new \n" "$name"
    continue
  fi
  file="$core_repo/Formula/$name.rb"
  deprecate_line=$(awk '/deprecate! date/ {print FNR}' "$file")
  if [ -z "$deprecate_line" ]; then
    printf "no deprecation in %s\n" "$name"
    continue
  fi
  if [ "$year_ago" -lt $(git log --no-patch --format=format:%ct -n1 -L"$deprecate_line,$deprecate_line:$file") ]; then
    printf "deprecation commit in %s is too new \n" "$name"
    continue
  fi

  sed -i.bak -E "s/deprecate! date: \"[0-9\-]*\"/disable! date: \"$date\"/" "$file"
  rm "$file.bak"

  git -C "$core_repo" add "$file"
  git -C "$core_repo" commit -m "$name: disable"
done
```

### CODE REVIEW

**Comment by carlocab** [comment]
Some of these formulae (e.g. `ykclient`) still have undeprecated dependents.

**Comment by SMillerDev** [comment]
Weird, I would have expected testbot to catch that.

File: Formula/gcc@4.9.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ class GccAT49 < Formula
   pour_bottle? only_if: :clt_installed
 
   # https://gcc.gnu.org/gcc-4.9/
-  deprecate! date: "2021-04-11", because: :deprecated_upstream
+  disable! date: "2022-06-17", because: :deprecated_upstream
```
Comment (Bo98) [inline Formula/gcc@4.9.rb:19]: GCC 4.9 support neeeds to be removed from Homebrew/brew before we do this.

File: Formula/apollo.rb [hunk 2]
```diff
@@ -19,7 +19,7 @@ class Apollo < Formula
   end
 
   # https://github.com/apache/activemq-apollo/commit/049d68bf3f94cdf62ded5426d3cad4ef3e3c56ca
-  deprecate! date: "2019-03-11", because: :deprecated_upstream
+  disable! date: "2022-06-17", because: :deprecated_upstream
```
Comment (cho-m) [inline Formula/apollo.rb:22]: There is another PR for this #103253

File: Formula/dep.rb [hunk 3]
```diff
@@ -18,7 +18,7 @@ class Dep < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "d7917c664bc0a540e065deb1e14671a5a818823672ec7a5a5caee34eb8feb664"
   end
 
-  deprecate! date: "2020-11-25", because: :repo_archived
+  disable! date: "2022-06-17", because: :repo_archived
```
Comment (cho-m) [inline Formula/dep.rb:21]: Used by `devd`, `ironcli` (deprecated), `vert` (deprecated)

File: Formula/dgraph.rb [hunk 4]
```diff
@@ -17,7 +17,7 @@ class Dgraph < Formula
     sha256 cellar: :any_skip_relocation, mojave:         "57fa2f974e4e0313fb58f7e35e2d9547a4d43319af7e84a4fc6619238e862885"
   end
 
-  deprecate! date: "2021-06-10", because: :unsupported
+  disable! date: "2022-06-17", because: :unsupported
```
Comment (cho-m) [inline Formula/dgraph.rb:20]: I wonder if we should make try to make this Linux-only & update to latest given this was deprecated due to upstream dropping macOS support.

Though, there may be an issue with behavior of macOS users who installed existing bottle. I think we saw a similar issue when `at-spi2-core` was resurrected as Linux-only.

File: Formula/ec2-api-tools.rb [hunk 5]
```diff
@@ -18,7 +18,7 @@ class Ec2ApiTools < Formula
   # Deprecated upstream somewhere between 2017-12-24 and 2018-09-09 here:
   # https://docs.aws.amazon.com/AWSEC2/latest/APIReference/Welcome.html
   # The current version (1.7.5.1) is from 2015-09-08.
-  deprecate! date: "2020-02-03", because: :deprecated_upstream
+  disable! date: "2022-06-17", because: :deprecated_upstream
```
Comment (cho-m) [inline Formula/ec2-api-tools.rb:21]: May be better to simultaneously disable with `aws-cfn-tools` and `elb-tools`. I deprecated those recently (#98492) so not sure if we want to wait a bit longer or just rely on original `ec2-api-tools` deprecation date.

File: Formula/glide.rb [hunk 6]
```diff
@@ -18,7 +18,7 @@ class Glide < Formula
   end
 
   # See: https://github.com/Masterminds/glide/commit/c64b14592409a83052f7735a01d203ff1bab0983
-  deprecate! date: "2021-01-02", because: :deprecated_upstream
+  disable! date: "2022-06-17", because: :deprecated_upstream
```
Comment (cho-m) [inline Formula/glide.rb:21]: Used by `jabba` and `textql`

File: Formula/protobuf@3.6.rb [hunk 7]
```diff
@@ -16,7 +16,7 @@ class ProtobufAT36 < Formula
 
   keg_only :versioned_formula
 
-  deprecate! date: "2021-02-19", because: :versioned_formula
+  disable! date: "2022-06-17", because: :versioned_formula
```
Comment (carlocab) [inline Formula/protobuf@3.6.rb:19]: Isn't this disabled already? Or did I forget to merge my disable PR?

Comment (carlocab) [reply Formula/protobuf@3.6.rb:19]: Oh, I see. This PR is older. 😅

File: Formula/isl@0.18.rb [hunk 8]
```diff
@@ -25,7 +25,7 @@ class IslAT018 < Formula
 
   keg_only :versioned_formula
 
-  deprecate! date: "2020-11-05", because: :versioned_formula
+  disable! date: "2022-06-17", because: :versioned_formula
```
[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/isl@0.18.rb:28]: gcc@5, ouch

**Comment by iMichka** [comment]
One thing that might go wrong here is that someone might have put a deprecation date far in the past, but you don't check when the deprecation was really added. So the 3 months delay between deprecation and disabling is maybe not respected for all formulae. Maybe looking at the date with "git blame" for that line would help?

**Comment by SMillerDev** [comment]
Rebuild the list/script taking the commit age into account.

File: Formula/sdl.rb [hunk 9]
```diff
@@ -65,7 +65,7 @@ class Sdl < Formula
   end
 
   # SDL 1.2 is deprecated, unsupported, and not recommended for new projects.
-  deprecate! date: "2013-08-17", because: :deprecated_upstream
+  disable! date: "2022-07-28", because: :deprecated_upstream
```
Comment (cho-m) [inline Formula/sdl.rb:68]: `sdl` is the most used deprecated formula. Until we figure out the migration plan to `sdl12-compat`/`sdl2`, this is probably going to stick around.

File: Formula/govendor.rb [hunk 10]
```diff
@@ -17,7 +17,7 @@ class Govendor < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "24497503629e520a1fe718029ab520b05c012677df065c9fd104afe4d898b8b8"
   end
 
-  deprecate! date: "2020-03-02", because: :repo_archived
+  disable! date: "2022-07-28", because: :repo_archived
```
Comment (cho-m) [inline Formula/govendor.rb:20]: `dockviz` still uses this to build. Last time we checked, the project was supposedly still maintained https://github.com/justone/dockviz/issues/49 and was going to migrate to go modules (comment from ~1 year ago)

File: Formula/guile@2.rb [hunk 11]
```diff
@@ -18,7 +18,7 @@ class GuileAT2 < Formula
 
   keg_only :versioned_formula
 
-  deprecate! date: "2020-04-07", because: :versioned_formula
+  disable! date: "2022-07-28", because: :versioned_formula
```
Comment (cho-m) [inline Formula/guile@2.rb:21]: `lilypond` still uses this.

File: Formula/ilmbase.rb [hunk 12]
```diff
@@ -18,7 +18,7 @@ class Ilmbase < Formula
   keg_only "ilmbase conflicts with `openexr` and `imath`"
 
   # https://github.com/AcademySoftwareFoundation/openexr/pull/929
-  deprecate! date: "2021-04-05", because: :unsupported
+  disable! date: "2022-07-28", because: :unsupported
```
Comment (cho-m) [inline Formula/ilmbase.rb:21]: Still need to figure out `ctl` and `field3d`. I was originally considering deprecating, but at least `ctl` has recently had commits. No work related to newer `imath`/`openexr` though.

**Comment by SMillerDev** [comment]
Gonna add a `brew uses --skip-deprecated` so I can let the script take that into account.

**Comment by carlocab** [comment]
Let me finish off https://github.com/Homebrew/brew/pull/12770. That'll probably help here.

File: Formula/jam.rb [hunk 13]
```diff
@@ -20,7 +20,7 @@ class Jam < Formula
   # The last Perforce release of Jam was version 2.6 in August of 2014. We will
   # keep the Perforce-controlled links and information posted here available
   # until further notice."
-  deprecate! date: "2021-07-10", because: :unmaintained
+  disable! date: "2022-07-28", because: :unmaintained
```
Comment (cho-m) [inline Formula/jam.rb:23]: `argyll-cms` and `lincity-ng` here.

File: Formula/vtk@8.2.rb [hunk 14]
```diff
@@ -17,7 +17,7 @@ class VtkAT82 < Formula
 
   keg_only :versioned_formula
 
-  deprecate! date: "2020-05-14", because: :versioned_formula
+  disable! date: "2022-07-28", because: :versioned_formula
```
Comment (cho-m) [inline Formula/vtk@8.2.rb:20]: And `gdcm` and `itk` here.

**Comment by carlocab** [comment]
Try your script again now.

File: Formula/opentsdb.rb [hunk 15]
```diff
@@ -16,7 +16,7 @@ class Opentsdb < Formula
     sha256 cellar: :any_skip_relocation, high_sierra: "5bcdc828069e124c16e1e6c8b2eb6732d0ef88533c27f60fcbb0bec369aca375"
   end
 
-  deprecate! date: "2020-11-13", because: :does_not_build
+  disable! date: "2022-07-31", because: :does_not_build
```
Comment (cho-m) [inline Formula/opentsdb.rb:19]: We are able to build `opentsdb` now (#100699) but I am stuck on figuring out why `hbase` LZO isn't working on Monterey (#104233). I could just remove compression or use gzip instead to get that re-enabled.

File: Formula/ec2-api-tools.rb [hunk 16]
```diff
@@ -18,7 +18,7 @@ class Ec2ApiTools < Formula
   # Deprecated upstream somewhere between 2017-12-24 and 2018-09-09 here:
   # https://docs.aws.amazon.com/AWSEC2/latest/APIReference/Welcome.html
   # The current version (1.7.5.1) is from 2015-09-08.
-  deprecate! date: "2020-02-03", because: :deprecated_upstream
+  disable! date: "2022-07-31", because: :deprecated_upstream
```
Comment (cho-m) [inline Formula/ec2-api-tools.rb:21]: For some formulae like `ec2-api-tools`, dependents were deprecated afterward (e.g. `aws-cfn-tools`, `elb-tools`).

Main question is how we want to deal with removal flow:
- we could disable dependents in this PR.
- we could keep them deprecated, maybe try to disable them before 1 year, and then finally remove dependency tree either at min/max of dates.

Comment (SMillerDev) [reply Formula/ec2-api-tools.rb:21]: I'm going to go for disabling them now. It's been more than 3 months already, so it's not conflicting with the policy.

**Review by iMichka — APPROVED** [review]

**Comment by iMichka** [comment]
Looks good; thanks for the update with the 3 month check.

I think it would be really great if we could move the script to a `brew` dev command if you have time. Or at least pin the code somewhere so that it can be used again.

**Pull request closed without merging**